nx/lib/nx.ex

```diff
@@ -6881,6 +6881,212 @@ defmodule Nx do
+
+  If the `:axis` option is given, it aggregates over
+  that dimension, effectively removing it. `axis: 0`
+  implies aggregating over the highest order dimension
+  and so forth. If the axis is negative, then the axis will
+  be counted from the back. For example, `axis: -1` will
+  always aggregate over the last dimension.
+
+  You may optionally set `:keep_axis` to true, which will
+  retain the rank of the input tensor by setting the reduced
+  axis to size 1.
+
+  ## Examples
+
+      iex> Nx.mode(Nx.tensor(42))
+      #Nx.Tensor<
+        s64
+        42
+      >
+
+      iex> Nx.mode(Nx.tensor([[1]]))
+      #Nx.Tensor<
+        s64
+        1
+      >
+
+      iex> Nx.mode(Nx.tensor([1, 2, 2, 3, 5]))
+      #Nx.Tensor<
+        s64
+        2
+      >
+
+      iex> Nx.mode(Nx.tensor([[1, 2, 2, 3, 5], [1, 1, 76, 8, 1]]))
+      #Nx.Tensor<
+        s64
+        1
+      >
+
+  ### Aggregating over an axis
+
+      iex> Nx.mode(Nx.tensor([[1, 2, 2, 3, 5], [1, 1, 76, 8, 1]]), axis: 0)
+      #Nx.Tensor<
+        s64[5]
+        [1, 1, 2, 3, 1]
+      >
+
+      iex> Nx.mode(Nx.tensor([[1, 2, 2, 3, 5], [1, 1, 76, 8, 1]]), axis: 1)
+      #Nx.Tensor<
+        s64[2]
+        [2, 1]
+      >
+
+      iex> Nx.mode(Nx.tensor([[[[1]]]]), axis: 1)
+      #Nx.Tensor<
+        s64[1][1][1]
+        [
+          [
+            [1]
+          ]
+        ]
+      >
+
+  ### Keeping axis
+
+      iex> Nx.mode(Nx.tensor([[1, 2, 2, 3, 5], [1, 1, 76, 8, 1]]), axis: 1, keep_axis: true)
+      #Nx.Tensor<
+        s64[2][1]
+        [
+          [2],
+          [1]
+        ]
+      >
+
+      iex> Nx.mode(Nx.tensor(1), keep_axis: true)
+      #Nx.Tensor<
+        s64[1]
+        [1]
+      >
+
+      iex> Nx.mode(Nx.tensor([[[1]]]), keep_axis: true)
+      #Nx.Tensor<
+        s64[1][1][1]
+        [
+          [
+            [1]
+          ]
+        ]
+      >
+
+      iex> Nx.mode(Nx.tensor([[[[1]]]]), axis: 1, keep_axis: true)
+      #Nx.Tensor<
+        s64[1][1][1][1]
+        [
+          [
+            [
+              [1]
+            ]
+          ]
+        ]
+      >
+  """
+  @doc type: :aggregation, from_backend: false
+  def mode(tensor, opts \\ []) do
+    opts = keyword!(opts, axis: nil, keep_axis: false)
+    %T{shape: shape, names: names} = tensor = to_tensor(tensor)
+
+    axis =
+      if opts[:axis] != nil,
+        do: Nx.Shape.normalize_axis(shape, opts[:axis], names),
+        else: opts[:axis]
+
+    tensor_rank = rank(tensor)
+    tensor_size = size(tensor)
+
+    cond do
+      tensor_rank == 0 ->
+        if opts[:keep_axis], do: new_axis(tensor, -1), else: tensor
+
+      tensor_size == 1 and axis == nil ->
+        if opts[:keep_axis], do: tensor, else: squeeze(tensor)
+
+      tensor_size == 1 and axis != nil ->
+        if opts[:keep_axis], do: tensor, else: squeeze(tensor, axes: [axis])
+
+      axis == nil ->
+        tensor = flatten(tensor)
+        mode_1d(tensor, keep_axis: opts[:keep_axis])
```

Comment:
If we flatten, with `keep_axis: true` we will return 1D, instead of the tensor dimensionality. We should optionally reshape into `Tuple.repeat(1, tensor_rank)`.

Also, for 1D we could use `mode_general(tensor, axis: 0)`, no? The implementation is pretty much the same other than general being vectorized, so there's likely no much overhead, and we avoid duplication, wdyt?